--- vcard.rb    2006-06-20 11:35:21.000000000 -0700
+++ /Users/jmeskill/vcard.rb    2006-06-20 11:34:39.000000000 -0700
@@ -953,6 +953,11 @@
 
     ## TITLE
 
+    # The TITLE value, a String.  A wrapper around #value('TITLE').
+    def title
+      value('TITLE')
+    end
+
     ## UID
 
     # The URL value, a Uri. A wrapper around #value('NOTE').
@@ -1167,6 +1172,24 @@
         self
       end
 
+      # Set the title field, TITLE.
+      #
+      # It can be set to a single String or an Array of String.
+      def title=(title)
+        delete_if { |l| l.name == 'TITLE' }
+        
+        @card << Vpim::DirectoryInfo::Field.create( 'TITLE', title );
+      end
+
+      # Set the org field, ORG.
+      #
+      # It can be set to a single String or an Array of String.
+      def org=(org)
+        delete_if { |l| l.name == 'ORG' }
+        
+        @card << Vpim::DirectoryInfo::Field.create( 'ORG', org );
+      end
+
       # Set the nickname field, NICKNAME.
       #
       # It can be set to a single String or an Array of String.

